googletrans/client.py

```diff
@@ -46,39 +52,85 @@ class Translator:
     :param raise_exception: if `True` then raise exception if smth will go wrong
+    :param http2: whether to use HTTP2 (default: True)
+    :param use_fallback: use a fallback method
     :type raise_exception: boolean
     """
 
-    def __init__(self, service_urls=None, user_agent=DEFAULT_USER_AGENT,
+    def __init__(self, service_urls=DEFAULT_CLIENT_SERVICE_URLS, user_agent=DEFAULT_USER_AGENT,
                  raise_exception=DEFAULT_RAISE_EXCEPTION,
                  proxies: typing.Dict[str, httpcore.SyncHTTPTransport] = None,
                  timeout: Timeout = None,
-                 http2=True):
+                 http2=True,
+                 use_fallback=False):
 
         self.client = httpx.Client(http2=http2)
         if proxies is not None:  # pragma: nocover
             self.client.proxies = proxies
 
         self.client.headers.update({
             'User-Agent': user_agent,
+            'Referer': 'https://translate.google.com',
         })
 
         if timeout is not None:
             self.client.timeout = timeout
 
-        self.service_urls = service_urls or ['translate.google.com']
-        self.token_acquirer = TokenAcquirer(
-            client=self.client, host=self.service_urls[0])
+        if use_fallback:
+            self.service_urls = DEFAULT_FALLBACK_SERVICE_URLS
+            self.client_type = 'gtx'
+            pass
+        else:
+            #default way of working: use the defined values from user app
+            self.service_urls = service_urls
+            self.client_type = 'tw-ob'
+            self.token_acquirer = TokenAcquirer(
+                client=self.client, host=self.service_urls[0])
+
         self.raise_exception = raise_exception
 
+    def _build_rpc_request(self, text: str, dest: str, src: str):
+        return json.dumps([[
+            [
+                RPC_ID,
+                json.dumps([[text, src, dest, True],[None]], separators=(',', ':')),
+                None,
+                'generic',
+            ],
+        ]], separators=(',', ':'))
+
     def _pick_service_url(self):
         if len(self.service_urls) == 1:
             return self.service_urls[0]
         return random.choice(self.service_urls)
 
-    def _translate(self, text, dest, src, override):
-        token = self.token_acquirer.do(text)
-        params = utils.build_params(query=text, src=src, dest=dest,
+    def _translate(self, text: str, dest: str, src: str):
+        url = urls.TRANSLATE_RPC.format(host=self._pick_service_url())
+        data = {
+            'f.req': self._build_rpc_request(text, dest, src),
+        }
+        params = {
+            'rpcids': RPC_ID,
+            'bl': 'boq_translate-webserver_20201207.13_p0',
+            'soc-app': 1,
+            'soc-platform': 1,
+            'soc-device': 1,
+            'rt': 'c',
+        }
+        r = self.client.post(url, params=params, data=data)
+
+        if r.status_code != 200 and self.raise_Exception:
```

Comment:
Should be `self.raise_exception` instead of `self.raise_Exception`

Reply:
@ssut